specutils/io/registers.py

```diff
@@ -51,7 +62,10 @@ def wrapper(*args, **kwargs):
         return wrapper
 
     def decorator(func):
-        io_registry.register_reader(label, dtype, func)
+        if _astropy_has_priorities():
```

Comment:
Is this check really necessary? If a user-defined reader/writer has no priority, seems we can just default to the lowest priority instead of resorting to reflection?

Reply:
Older versions of `io_registry.register_reader` won't accept the additional argument, so if we don't do this check, we'll get exceptions when trying to add a priority. Once all supported versions of astropy accept priorities, we can drop the check.